packages/compass-crud/src/components/document-footer.jsx

```diff
@@ -247,9 +260,12 @@ DocumentFooter.displayName = 'DocumentFooter';
 DocumentFooter.propTypes = {
   doc: PropTypes.object.isRequired,
   replaceDocument: PropTypes.func.isRequired,
-  updateDocument: PropTypes.func.isRequired,
+  updateDocument: PropTypes.func,
   cancelHandler: PropTypes.func,
-  api: PropTypes.any
+  api: PropTypes.any,
+  editing: PropTypes.bool,
+  modified: PropTypes.bool,
+  containsErrors: PropTypes.bool
```

Comment:
This sorta allows for controlled / uncontrolled behavior for this footer component which is still probably messier than it should be, but I'll be tackling footer extraction separately so feels like this implementation is okay for now (but please tell me if you think that it's not)